src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java

```diff
@@ -0,0 +1,77 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.util.TeleportToColony;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.minecolonies.coremod.commands.AbstractSingleCommand.Commands.HOMETP;
+
+/**
+ * this command is made to TP a player to their home colony.
+ */
+public class HomeTeleportCommand extends AbstractSingleCommand
+{
+    public static final  String DESC             = "home";
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public HomeTeleportCommand(@NotNull final String... parents)
+    {
+        super(parents);
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "home";
+    }
+
+    @Override
+    public void execute(@NotNull MinecraftServer server, @NotNull ICommandSender sender, @NotNull String... args) throws CommandException
+    {
+
+        //see if player is allowed to use in the configs
+            if (!canCommandSenderUseCommand(HOMETP))
+            {
+                sender.getCommandSenderEntity().addChatMessage(new TextComponentString("This is not allowed on this server."));
+                return;
+            }
+            else
+            {
+                //send the info to the Colony TP utils
+                String homeID = "99999";
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'homeID' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)